excel-export-service/src/apiError.ts

```diff
@@ -0,0 +1,15 @@
+import httpStatus = require("http-status");
+import { ExtendableError } from "./errors";
+
+/** * Class representing an API error. *
+ * @extends ExtendableError
+ * @param {string} message - Error message.
+ * @param {number} status - HTTP status code of error.
+ * @param {boolean} isPublic - Whether the message should be visible to user or not.
+ * @param {string | null} originalMessage - Original Error message that was thrown.
+ */
+export class APIError extends ExtendableError {
```

Comment:
Q: Why not extend Error directly?

Reply:
@SamuelPull It might be useful for more granular error management. E.g. if app is communicating not just via Rest api, but also sockets or Kafka messages, which is not the case in this instance, so I removed the ExtendableError to simplify the structure.

Reply:
It's hard to always distinguish between a thoughtful anticipation of changing requirements, and littering the code with unnecessary design elements. Maybe the future will tell that this class hierarchy was a better choice.